monai/networks/layers/utils.py

```diff
@@ -124,3 +126,18 @@ def get_pool_layer(name: tuple | str, spatial_dims: int | None = 1):
     pool_name, pool_args = split_args(name)
     pool_type = Pool[pool_name, spatial_dims]
     return pool_type(**pool_args)
+
+
+def get_rel_pos_embedding_layer(name: tuple | str, s_input_dims: Optional[tuple], c_dim: int, num_heads: int):
+    embedding_name, embedding_args = split_args(name)
+    kw_args = dict(embedding_args)
+    embedding_type = RelPosEmbedding[embedding_name]
+
+    if has_option(embedding_type, "s_input_dims") and "s_input_dims" not in kw_args:
+        embedding_args["s_input_dims"] = s_input_dims
+    if has_option(embedding_type, "c_dim") and "c_dim" not in kw_args:
+        embedding_args["c_dim"] = c_dim
+    if has_option(embedding_type, "num_heads") and "num_heads" not in kw_args:
+        embedding_args["num_heads"] = num_heads
+
+    return embedding_type(**embedding_args)
```

Comment:
```suggestion
def get_rel_pos_embedding_layer(name: tuple | str, s_input_dims: Optional[tuple], c_dim: int, num_heads: int):
    embedding_name, embedding_args = split_args(name)
    embedding_type = RelPosEmbedding[embedding_name]
    # create a dictionary with the default values which can be overridden by embedding_args
    kw_args = {"s_input_dims": s_input_dims, "c_dim": c_dim, "num_heads": num_heads, **embedding_args}
    # filter out unused argument names
    kw_args = {k: v for k, v in kw_args.items() if has_option(embedding_type, k)}
            
    return embedding_type(**kw_args)
```
I think this does what you wanted your implementation to do, unless I'm mistaken there was some issue with your logic when you're checking if a value is *not* in `kw_args` and adding it to `embedding_args` if that was true. In this case you wouldn't use the value `kw_args` if it was present.

Reply:
Yes, you're absolutely right. I commit your suggestion